packages/app-accounts/src/UploadButton.tsx

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017-2018 @polkadot/app-accounts authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { BareProps, I18nProps } from '@polkadot/ui-app/types';
+import { KeyringPair$Json } from '@polkadot/util-keyring/types';
+
+import React from 'react';
+import isUndefined from '@polkadot/util/is/undefined';
+import keyring from '@polkadot/ui-keyring/index';
+import arrayContainsArray from '@polkadot/ui-app/util/arrayContainsArray';
+import File from '@polkadot/ui-app/Params/Param/File';
+
+import UploadModal from './UploadModal';
+import translate from './translate';
+
+type State = {
+  address: string,
+  password: string,
+  isPasswordModalOpen: boolean,
+  error?: React.ReactNode,
+  uploadedFileKeyringPair: KeyringPair$Json | undefined
+};
+
+type Props = I18nProps & BareProps & {
+  onChangeAccount: any
+};
+
+class UploadButton extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = this.emptyState();
+  }
+
+  handleUploadedFiles = (fileBytes: Uint8Array): void => {
+    if (!fileBytes || !fileBytes.length) {
+      console.error('Error retrieving file list');
+      return;
+    }
+
+    const blob = new Blob([fileBytes], { type: 'text/plain;charset=utf-8' });
+    const fileReader = new FileReader();
+
+    fileReader.onload = (e) => {
+      try {
+        if (!isUndefined(e) && e.target !== null) {
+          // Cast to type 'any' since property 'result' does exist on type 'EventTarget'
+          // Reference: https://stackoverflow.com/a/45017155/3208553
+          const fileContents: any = JSON.parse((e.target as any).result);
+          const expectedJsonProperties = ['address', 'encoding', 'meta'];
```

Comment:
Hoist these constants to file-level. (i.e. it does not need to be created as a new variable each time the function is executed)